scipy/stats/tests/test_morestats.py

```diff
@@ -2458,108 +2461,81 @@ class TestCircFuncs:
     # var.circular(x)
     @pytest.mark.parametrize("test_func,expected",
                              [(stats.circmean, 0.167690146),
-                              (stats.circvar, 0.006455174270186603),
+                              (stats.circvar, 0.006455174000787767),
                               (stats.circstd, 6.520702116)])
-    def test_circfuncs(self, test_func, expected):
-        x = np.array([355, 5, 2, 359, 10, 350])
-        assert_allclose(test_func(x, high=360), expected, rtol=1e-7)
+    def test_circfuncs(self, test_func, expected, xp):
+        x = xp.asarray([355., 5., 2., 359., 10., 350.])
+        xp_assert_close(test_func(x, high=360), xp.asarray(expected))
 
-    def test_circfuncs_small(self):
-        x = np.array([20, 21, 22, 18, 19, 20.5, 19.2])
-        M1 = x.mean()
+    def test_circfuncs_small(self, xp):
+        x = xp.asarray([20, 21, 22, 18, 19, 20.5, 19.2], dtype=xp.float64)
```

Comment:
How come we can't use the default floating dtype here? It might be useful to add a comment when we specify a `dtype` so that looking back it is clear that it a deliberate choice

Reply:
The `circvar` reference value was inaccurate enough that fixing it would be enough to not require `float64`. I'll look into other options or add a comment.

Reply:
Added a comment.